File: VB/ExportToDataTableWorkbookExample/Form1.vb
Imports DevExpress.Spreadsheet
Imports DevExpress.Spreadsheet.Export
Imports System.Data
Imports System.Windows.Forms

Namespace ExportToDataTableWorkbookExample
    Partial Public Class Form1
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Private ds As DataTable
        Public Sub New()
            InitializeComponent()

        End Sub

        Private Sub barBtnExport_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles barBtnExport.ItemClick
            If ds IsNot Nothing Then
                Return
            End If
'            #Region "#exportdatatable"
            Dim wbook As New Workbook()
            wbook.LoadDocument("TopTradingPartners.xlsx")
            Dim worksheet As Worksheet = wbook.Worksheets(0)
            Dim range As Range = worksheet.Tables(0).Range

            Dim dataTable As DataTable = worksheet.CreateDataTable(range, True)
            ' Change the data type of the "As Of" column to text.
            dataTable.Columns("As Of").DataType = System.Type.GetType("System.String")

            Dim exporter As DataTableExporter = worksheet.CreateDataTableExporter(range, dataTable, True)
            AddHandler exporter.CellValueConversionError, AddressOf exporter_CellValueConversionError
            Dim myconverter As New MyConverter()
            exporter.Options.CustomConverters.Add("As Of", myconverter)
            ' Set the export value for empty cell.
            myconverter.EmptyCellValue = "N/A"
            exporter.Options.ConvertEmptyCells = True

            exporter.Options.DefaultCellValueToColumnTypeConverter.SkipErrorValues = False

            exporter.Export()
'            #End Region ' #exportdatatable
            ds = dataTable
            gridControl1.DataSource = ds
        End Sub
        #Region "#CellValueConversionError"
        Private Sub exporter_CellValueConversionError(ByVal sender As Object, ByVal e As CellValueConversionErrorEventArgs)
            MessageBox.Show("Error in cell " & e.Cell.GetReferenceA1())
            e.DataTableValue = Nothing
            e.Action = DataTableExporterAction.Continue
        End Sub
        #End Region ' #CellValueConversionError
    End Class
End Namespace

File: CS/ExportToDataTableWorkbookExample/Form1.cs
using DevExpress.Spreadsheet;
using DevExpress.Spreadsheet.Export;
using System.Data;
using System.Windows.Forms;

namespace ExportToDataTableWorkbookExample {
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm {
        DataTable ds;
        public Form1() {
            InitializeComponent();
            
        }

        private void barBtnExport_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            if (ds != null) return;
            #region #exportdatatable
            Workbook wbook = new Workbook();
            wbook.LoadDocument("TopTradingPartners.xlsx");
            Worksheet worksheet = wbook.Worksheets[0];
            Range range = worksheet.Tables[0].Range;

            DataTable dataTable = worksheet.CreateDataTable(range, true);
            // Change the data type of the "As Of" column to text.
            dataTable.Columns["As Of"].DataType = System.Type.GetType("System.String");
           
            DataTableExporter exporter = worksheet.CreateDataTableExporter(range, dataTable, true);
            exporter.CellValueConversionError += exporter_CellValueConversionError;
            MyConverter myconverter = new MyConverter();
            exporter.Options.CustomConverters.Add("As Of", myconverter);
            // Set the export value for empty cell.
            myconverter.EmptyCellValue = "N/A";
            exporter.Options.ConvertEmptyCells = true;
            
            exporter.Options.DefaultCellValueToColumnTypeConverter.SkipErrorValues = false;

            exporter.Export();
            #endregion #exportdatatable
            ds = dataTable;
            gridControl1.DataSource = ds;
        }
        #region #CellValueConversionError
        void exporter_CellValueConversionError(object sender, CellValueConversionErrorEventArgs e) {
            MessageBox.Show("Error in cell " + e.Cell.GetReferenceA1());
            e.DataTableValue = null;
            e.Action = DataTableExporterAction.Continue;
        }
        #endregion #CellValueConversionError
    }
}
